atlasdb-api/src/main/java/com/palantir/atlasdb/keyvalue/api/KeyValueService.java

```diff
@@ -680,4 +681,21 @@ void createTable(@QueryParam("tableRef") TableReference tableRef, byte[] tableMe
     @Path("node-availability-status")
     @Consumes(MediaType.APPLICATION_JSON)
     ClusterAvailabilityStatus getClusterAvailabilityStatus();
+
+    /**
+     * Some KVSs may provide a logging arg supplier, which knows whether constructs are
+     * safe for logging (e.g. a table name or column name). This may need to be rehydrated
+     * if there is a change to the schema that possibly results in some construct becoming
+     * safe or unsafe to log.
+     */
+    default void rehydrateLoggingArgSupplier() {
+        // no op
```

Comment:
Unfortunately, there are implementations of KVS that don't extend from `AbstractKVS` (including one in large internal product that's used in logging). I was trying to avoid an API break here.